server/accounts.go

```diff
@@ -2623,6 +2629,35 @@ func (s *Server) UpdateAccountClaims(a *Account, ac *jwt.AccountClaims) {
 			}
 		}
 	}
+
+	// if we encountered issues during this process, retry in a bit
```

Comment:
This feels like it could possibly spin no?

Reply:
Like I mentioned, we could always keep track of the number of times we tried and give up after a certain amount.

Reply:
If there is an import that fails I think that means the import is invalid but does not mean anything about the account itself. Should we be trying to reload?

Reply:
or it means the account exporting has not been updated (to add the export) yet.

Reply:
If this becomes an issue we can add a limit as @kozlovic said. When we do that we probably should also evict the account. (smth yet to be added)

Reply:
Can't add an import without the export. This is more the import required a token and the token expired or was revoked, or the exporting account was updated to remove or change the export.

Reply:
I don't think if an import fails the account should be evicted at all. That would mean the user can not connect at all. We would want to make it easy for the importing account to debug though.

Reply:
If you modify two accounts (add an export and corresponding import) and push all, because the server has to process one of them first you have a high likelihood of that happening

Reply:
Hmm, good to think about the race condition for sure. My hope would be the export account update gets pushed first but I hear what you are saying.

Reply:
> I don't think if an import fails the account should be evicted at all. That would mean the user can not connect at all. We would want to make it easy for the importing account to debug though.

The problem I see with that is noticing that smth is wrong in the first place. 
This particular error manifests itself as missing messages. 
Events associated with disconnect/reconnect are better observable than the lack of messages.

Reply:
Are we comfortable though with an account being suspended due to a bad import until said import is removed or fixed and the account re-pushed? That sounds very dangerous to me and violates path of least surprise.

Reply:
We don't expect this condition to happen a lot, so as of now we keep trying. 
Do you think this is sufficient for now?


> Are we comfortable though with an account being suspended due to a bad import until said import is removed or  fixed and the account re-pushed? That sounds very dangerous to me and violates path of least surprise.

The account wouldn't be suspended. Clients would be able to do everything while they are connected. 
Every once in a while, they would be disconnected with a close reason account import broken.
Perhaps we could only close clients with subscriptions for the import, as well as the ones publishing to it.
However, this seems to be a discussion to be had when we get there.

Reply:
I don't think we should disconnect clients at all. We could keep trying to resolve but feels like we should be smarter. For instance we should know if an account has a broken import on it, and when we get an export update go try to reload the import account that was broke again then.

Reply:
I take this as don't use timers then

Reply:
I would not auto try to resolve TBH, would pick concrete event that would cause us to try to resolve TBH.

Reply:
We went down this route as we do not expect adding imports/exports happens often.
So automatically retrying once a second per initial failure appeared more straight forward than adding and maintaining data structures to track the issue.
An edge case to consider for an event driven approach is the update of the exporting account never happening. (say the import was wrong and gets removed)

Reply:
Do we have tests that show we get an import account before and export account update and produce those failures?

I don't think we should keep trying to reload or disconnect existing clients due to import failure. Just my $0.02.

Reply:
I will let Matthias comment, but he has been investigating an issue with a user that shows interest drop that has to do with import/export and accounts push. He did add tests to this PR. I do think the PR is fine as it is, the only thing we could do is stop trying after some time, but just giving up without closing clients seem wrong because then users would have the interest-loss issue which is difficult to debug as opposed to get the client disconnected.

Reply:
If I import a service and someone revokes my token, I don't think that should have all my connections cut off. I agree we need more tools for end users to be able to identify the issue/

Reply:
I have been contacted about the same symptom multiple times.
Because there was such a delay between dropping the subscriptions and investigating, logs turned out to be of no use.
I realized what might be going on based on narrowing down the situation to a single server + accounts with imports and then reading code.
This PR contains unit tests for fetch failing as well as unfortunate update reordering.
I'm very interested in this becoming part of the next release just so I know if I'm on the right track. 
After-all this still may be another issue altogether.

I just pushed two commits.
First one keeps track of the account names experiencing the issue.
After this was done (passed all tests) I realized that ranging over all accounts is smth that is done, and could slim down the necessary state.

So I'd say have a look at the last commit. 
Only if that does not fit the bill, have a look at the previous one.

Reply:
ok I think we are closer. Could you squash these into 1 and ping me and will take another look all together. Thx!

Reply:
Done